uPortSDK/Classes/Crypto/Crypto.swift

```diff
@@ -0,0 +1,172 @@
+//
+//  Crypto.swift
+//  uPortSDK
+//
+
+import Foundation
+import Sodium
+import Clibsodium
+
+/**
+ * This struct exposes methods to encrypt and decrypt messages according to the uPort spec at
+ * https://github.com/uport-project/specs/blob/develop/messages/encryption.md
+ */
+public struct Crypto
+{
+    public struct Constants
+    {
+        static let asyncEncryptionAlgorithm = "x25519-xsalsa20-poly1305"
+        static let blockSize = 64.0
+    }
+    /**
+     * This class encapsulates an encrypted message that was produced using
+     * https://github.com/uport-project/specs/blob/develop/messages/encryption.md
+     */
+    public struct EncryptedMessage: Codable
+    {
+        var version: String = Constants.asyncEncryptionAlgorithm
+        var nonce: String
+        var ephemPublicKey: String
+        var ciphertext: String
+        
+        
+        public init(nonce: String, ephemPublicKey: String, ciphertext: String)
+        {
+            self.nonce = nonce
+            self.ephemPublicKey = ephemPublicKey
+            self.ciphertext = ciphertext
+        }
+        
+        public func toJson() -> Data
+        {
+            return try! Data(JSONEncoder().encode(self))
+        }
+        
+        public static func fromJson(jsonData: Data) -> EncryptedMessage
+        {
+            return try! JSONDecoder().decode(EncryptedMessage.self, from: jsonData)
+        }
+    }
+    
+    /**
+     Calculates the encryption public key corresponding to the secret key.
+     
+     - Parameter secretKey: The base64 encoding string secret key
+     - Returns: A base64 encoded public key
+     */
+    public static func getEncryptionPublicKey(secretKey: String) -> String?
+    {
+        let secretKeyDecoded = secretKey.decodeBase64()
+        let skBytes = Bytes(secretKeyDecoded)
+        guard skBytes.count == 32 else {
+            print("Secret Key must be 32 bytes")
```

Comment:
Is returning `nil` appropriate?  Wouldn't throwing an exception be better?  All depends on under what circumstances this occurs: If it's a 'normal' situation (so with valid input and all) that there's no result, then `nil` is fine, otherwise if `nil` is only when there're something wrong then `throw` w/could be appropriate.